lib/ecto/repo.ex

```diff
@@ -498,6 +506,35 @@ defmodule Ecto.Repo do
               {:ok, Ecto.Model.t} | {:error, Ecto.Changeset.t}
 
   @doc """
+  Inserts or updates a model or a changeset depending on whether the model
+  was loaded from the database or created by the user. This is especially
+  helpful when accepting user-provided ids (the use of UUIDs is highly
+  recommended in this scenario!).
+
+  Please note that for this to work, you will have to load existing models from
+  the database. So even if the model exists, this won't work:
+
+      model = %Post{id: 'existing_id', ...}
+      MyRepo.insert_or_update model
+      # => {:error, "id already exists"}
+
+  ## Example
+
+      case MyRepo.get(Post, id) do
+        nil  -> %Post{id: id} # Post not found, we create a new one
+        post -> post          # Post exists, let's use it
+      end
+      |> Post.changeset(changes)
+      |> MyRepo.insert_or_update
+      |> case do
+        {:ok, model}        -> # Inserted or updated with success
+        {:error, changeset} -> # Something went wrong
+      end
```

Comment:
Remove:
> "(the use of UUIDs is highly  recommended in this scenario!)."

Is valid also for generated integers from sequences. (Some apps use one global sequence)